packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmDictionaryTests.kt

```diff
@@ -0,0 +1,619 @@
+    private lateinit var realm: Realm
+
+    private val managedTesters: List<DictionaryApiTester<*, RealmDictionaryContainer>> by lazy {
+        // TODO use mapNotNull to introduce each supported type progressively and return null for
+        //  those types that remain unsupported - remove this hack when all types are supported
+        descriptors.mapNotNull {
+            val elementType = it.elementType
+            when (val classifier = elementType.classifier) {
+                RealmAny::class -> null
+                RealmObject::class -> RealmObjectDictionaryTester(
+                    realm,
+                    getTypeSafety(
+                        classifier,
+                        elementType.nullable
+                    ) as DictionaryTypeSafetyManager<RealmDictionaryContainer>,
+                    classifier
+                )
+                ByteArray::class -> ByteArrayTester(
+                    realm,
+                    getTypeSafety(
+                        classifier,
+                        elementType.nullable
+                    ) as DictionaryTypeSafetyManager<ByteArray>,
+                    classifier
+                )
+                else -> GenericDictionaryTester(
+                    realm,
+                    getTypeSafety(classifier, elementType.nullable),
+                    classifier
+                )
+            }
+        }
+    }
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(
+            schema = setOf(RealmDictionaryContainer::class)
+        ).directory(tmpDir)
+            .build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::realm.isInitialized && !realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun unmanaged() {
+        // No need to be exhaustive here, just checking delegation works
+        val dictionary = realmDictionaryOf<RealmDictionaryContainer>()
+        val entry = Pair("A", RealmDictionaryContainer().apply { stringField = "Dummy" })
+        assertTrue(dictionary.isEmpty())
+        dictionary["A"] = entry.second
+        assertEquals(1, dictionary.size)
+        assertEquals(entry.second.stringField, assertNotNull(dictionary[entry.first]).stringField)
+    }
+
+    @Test
+    fun realmMapEntryOf() {
+        // TODO will be added when support for dictionary.entries is added
+    }
+
+    @Test
+    fun realmDictionaryInitializer_realmDictionaryOf() {
+        // No need to be exhaustive here
+        val realmDictionaryFromArgsEmpty: RealmDictionary<String> = realmDictionaryOf()
+        assertTrue(realmDictionaryFromArgsEmpty.isEmpty())
+
+        val args = listOf("A" to "1", "B" to "2").toTypedArray()
+        val realmDictionaryFromArgs: RealmDictionary<String> = realmDictionaryOf(*args)
```

Comment:
Should we also allow a `Collection<RealmMapEntry>` as input to `realmDictionaryOf`...having to use the spread operator here seems a bit wasteful

Reply:
I added a new factory function that receives a collection of pairs.